### ISSUE TITLE
Missing general `preferredName` for some GND entries

### ISSUE THREAD

**Issue by fsteeg** [hbz/lobid#84]
Usually all GND entries have a general `preferredName` property in addition to the specialized `preferredNameForTheWhatever` property. These are missing for some entries, e.g.

http://api.lobid.org/subject?id=4023168-9&format=full
http://api.lobid.org/subject?id=4019947-2&format=full
http://api.lobid.org/subject?id=4007862-0&format=full

**Comment by dr0i** [comment]
Interesting. We don't change any data in gnd so this should be a thing on provider side. Also interesting for [lobid/lodmill#506](https://github.com/lobid/lodmill/issues/506#issuecomment-68706192).

**Comment by dr0i** [comment]
... we don't change the data besides some inferencing, see lobid/lodmill#354. Maybe there's the problem.

**Comment by acka47** [comment]
Obviously, this is no problem in the GND, e.g. `rapper -o turtle http://d-nb.info/gnd/4023168-9` delivers:

```
@prefix gndo: <http://d-nb.info/standards/elementset/gnd#> .


<http://d-nb.info/gnd/4023168-9>
   ...
    gndo:preferredNameForThePlaceOrGeographicName "Hamm (Westf)" ;
    gndo:variantNameForThePlaceOrGeographicName "Bad Hamm", "Hamm <Westfalen>", "Hamm <Westfalen> / Abteilung Statistik", "Hamm <Westfalen> / Abteilung für Wirtschaft, Verkehr und Statistik", "Hamm <Westfalen> / Hauptamt", "Hamm <Westfalen> / Oberbürgermeister", "Hamm <Westfalen> / Oberstadtdirektor", "Hamm <Westfalen> / Presse- und Werbeamt", "Hamm <Westfalen> / Pressestelle", "Hamm <Westfalen> / Stadtverwaltung", "Hauptamt <Hamm, Westfalen>", "Presse- und Werbeamt <Hamm, Westfalen>", "Stadtverwaltung <Hamm, Westfalen>" ;
    a gndo:TerritorialCorporateBodyOrAdministrativeUnit ;
    foaf:page <http://de.wikipedia.org/wiki/Hamm> .
```

**Comment by acka47** [comment]
I had a look at some examples. Obviously this doesn't have anything to do with the entities' type. We hae CorporateBody (http://lobid.org/subject?id=2085022-0, http://lobid.org/subject?id=2153289-8, http://lobid.org/subject?id=2164163-8, http://lobid.org/subject?id=16070317-7), TerritorialCorporateBodyOrAdministrativeUnit  (http://lobid.org/subject?id=4005044-0, http://lobid.org/subject?id=4005406-8), SubjectHeadingSensoStricto (http://lobid.org/subject?id=4002728-4), DifferentiatedPerson (http://lobid.org/subject?id=118515470, http://lobid.org/subject?id=11894519X, http://lobid.org/subject?id=121705668) and probably others affected.

**Comment by acka47** [comment]
I will look next Monday whether these are always the same entities affected by this or whether the entities affected vary after each indexing.

**Comment by acka47** [comment]
> I will look next Monday whether these are always the same entities affected by this or whether the entities affected vary after each indexing.

@dr0i will have to initiate the re-indexing of GND entities including inferencing of `preferedName` information. Than we will have another look.

**Comment by acka47** [comment]
BTW, this was also reported by NWBib staff, see https://wiki1.hbz-nrw.de/x/CoCDAw, 5.c.

**Comment by acka47** [comment]
@dr0i told me that updated GND data is fully re-indexed once a week. I compared the resources with missing labels in NWBib to thise from March 16 and both sets of GND resources with missing labels was identical. This has to be a systematic error, then.

**Comment by acka47** [comment]
> ... we don't change the data besides some inferencing, see lobid/lodmill#354. Maybe there's the problem.

As I haven't said it clearly yet: The problem must lie at the inferencing part. Shall we re-open lobid/lodmill#354?

**Comment by dr0i** [comment]
I cannot find `preferredName` in e.g. `rapper -o turtle http://d-nb.info/gnd/4023168-9`. So this is a provider issue? Were they to be expected to deliver the  `preferredName` at all or is this an inferencing task?

**Comment by acka47** [comment]
> I cannot find `preferredName` in e.g. `rapper -o turtle http://d-nb.info/gnd/4023168-9`. So this is a provider issue? 

No.

> Were they to be expected to deliver the  `preferredName` at all or is this an inferencing task?

No , they weren't, we do this with inferencing. As a `preferredNameForThePlaceOrGeographicName` exists, `preferedName` should actually be inferenced, but it isn't.

As I said in https://github.com/hbz/lobid/issues/84#issuecomment-81704584, this doesn't seem to have anything to do with the specific type of an entity but it does affect the same resources with every new inferencing/indexing and must be a systematic error (https://github.com/hbz/lobid/issues/84#issuecomment-88103538).

**Comment by dr0i** [comment]
With #139 (not using hadoop anymore to enrich with gnd data) the `preferredName` is resolved via hbz catalog directly and thus present in the lobid resources endpoint (e.g. 
http://lobid.org/resource/HT013468451).
 
I acknowledge that the principle issue remains for the subject endpoint (e.g. http://api.lobid.org/subject?id=4023168-9&format=full) but ask if this issue is of some pressure or is it rather that we don't make use of it in any service anyhow ? Would then want to have postponed it into backlog.

**Comment by acka47** [comment]
> I acknowledge that the principle issue remains for the subject endpoint (e.g. http://api.lobid.org/subject?id=4023168-9&format=full) but ask if this issue is of some pressure or is it rather that we don't make use of it in any service anyhow ? Would then want to have postponed it into backlog.

We use this in the subkect facets of NWBib. Take a look at http://lobid.org/nwbib/search?location=&q=* and open the `Schlagwörter` facet and you will see on top the subjects without a general `preferredName`. That's why we will have to resolve this in some point of the enar future.

**Comment by dr0i** [comment]
After discussion with @acka47 : 
we should not use the subject endpoint (which yields i.a. non normalized data) but the resource endpoint. In the latter, we must provide an array of preferredNames . This can be done by:

a) create this array in addition
b) by providing a framed json-ld with the subjects as an array

Whilst b) may break the API (especially when json-ld is 'compact'ed ) we could begin with building a API2.0 index in parallel (which then can be used by NWBib). a) would work but is just dumb workaround. We will discuss further more next monday.

**Comment by dr0i** [comment]
@acka47 @fsteeg and @dr0i discussed this today. We choose variant b).
This ticket is postponed until https://github.com/hbz/lobid-resources/issues/1 is solved.

**Comment by acka47** [comment]
For two of the three resources named in the description of this ticket, by now `preferredName` information exists. See http://lobid.org/subject?id=4023168-9&format=full & http://lobid.org/subject?id=4023168-9&format=full. Obviously, the resources that are affected do change.

**Comment by dr0i** [comment]
Going [back to the status quo of discussion](https://github.com/hbz/lobid/issues/84#issuecomment-133347115) I think we can close this issue as we have an subject array in Api 2.0. @acka47 @fsteeg ?

**Comment by acka47** [comment]
Hmm, actually the problem in the subject endpoint still exists. As nobody else complained than we ourselves, we might close it anyway.

**Comment by fsteeg** [comment]
Only the third example from the original comment still seems to have the problem. And we don't use this ourselves. I don't think we'll put further work into the 1.x APIs in general. So yes, I think we can close it.
